java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/AnalysisResult.java

```diff
@@ -92,11 +98,49 @@ private <T> void formatLog(VersionKey root, Map<VersionKey, List<T>> map, String
   }
 
   private String beginSeparator(VersionKey versionKey, VersionKey root) {
-    return String.format("====================== %s, dependency of %s ======================",
-        versionKey.toString(), root.toString());
+    String relation = versionKey.equals(root) ? "self" : "dependency";
+    return String.format("====================== %s, %s of %s ======================",
+        versionKey, relation, root);
   }
 
   private String endSeparator() {
     return "===========================================================";
   }
+
+  private String packageInfoReport() {
```

Comment:
Would you add a test case of the pull request description?


Java 17's test block should make it easy. https://blogs.oracle.com/javamagazine/post/text-blocks-come-to-java

Reply:
> Would you add a test case of the pull request description?

It's not easily test using unit test as the encoded string needs to be sent to deps.dev to verify the correctness.

I'll write an integration test for it.

Reply:
Would you write this in a way that the formatter function should operate on a Record class to generate the formatted string.

In this design, the unit test for the formatter does not need to access deps.dev.

Reply:
Wrote unit tests to verify the encoded string.